### PR TITLE
HHH-11540 - Test and fix

### PR DESCRIPTION
**Pull request by beikov**
https://hibernate.atlassian.net/browse/HHH-11540

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/metamodel/internal/AttributeFactory.java [hunk 1]
```diff
@@ -213,7 +213,7 @@ public Property getPropertyMapping() {
 			case EMBEDDABLE: {
 				final Component component = (Component) typeContext.getValue();
 				final EmbeddableTypeImpl<Y> embeddableType = new EmbeddableTypeImpl<Y>(
-						typeContext.getBindableType(),
+						component.getComponentClass(),
```
Comment (dreab8) [inline hibernate-core/src/main/java/org/hibernate/metamodel/internal/AttributeFactory.java]: with dynamic map components the value of `componentClassName` is null so `component.getComponentClass()` throws an Exception. 
I think the code should be changed to
```
Class javaType = null;
if ( component.getComponentClassName() == null ) {
      javaType = typeContext.getBindableType();
}
else {
      javaType = component.getComponentClass();
}
final EmbeddableTypeImpl<Y> embeddableType = new EmbeddableTypeImpl<Y>(
	javaType,
	typeContext.getAttributeMetadata().getOwnerType(),
	(ComponentType) typeContext.getValue().getType()
);
```

**Comment by beikov** [comment]
I updated the PR to what you suggested.

**Comment by dreab8** [comment]
@beikov Thanks

**Comment by dreab8** [comment]
applied upstream

**Pull request closed without merging**